src/composables/useApiGetMerchandise.ts

```diff
@@ -111,8 +126,205 @@ export const useApiGetMerchandise = (
+   * where each product (ID) holds information about its available variants.
+   */
+  const computeMerchandiseItems = (): MerchandiseItem[] => {
+    logger?.info('Computing merchandise items');
+    logger?.debug(`Transform <${merchandise.value.length}> items`);
+
+    // transform items grouped by name into MerchandiseItem
+    const items = merchandise.value.map((item): MerchandiseItem => {
+      // get all items with the same name (variants)
+      const variants = merchandiseGroupedByName.value[item.name];
+      logger?.debug(`Item <${item.name}> has <${variants.length}> variants`);
+
+      /**
+       * Get all unique genders available for this item name and size.
+       * These will determine the option values for the gender select.
+       */
+      const genderOptions: FormOption[] = [
+        ...new Set(
+          variants
+            .filter((variant) => variant.size === item.size)
+            .map((variant) => variant.sex),
+        ),
+      ].map((sex) => ({
+        label: getGenderLabel(sex),
+        value: sex,
+      }));
+      logger?.debug(
+        `Gender options <${JSON.stringify(genderOptions, null, 2)}>`,
+      );
+
+      /**
+       * Get all sizes available for this item name and gender.
+       * These will determine the option values for the size select.
+       */
+      const sizeOptions: FormOption[] = variants
+        .filter((variant) => variant.sex === item.sex)
+        .map((variant) => ({
+          label: variant.size,
+          value: variant.id,
+        }));
+      logger?.debug(`Size options <${JSON.stringify(sizeOptions, null, 2)}>`);
+
+      // create image array from t_shirt_preview
+      const images: Image[] = [
+        {
+          src: item.t_shirt_preview,
+          alt: item.name,
+        },
+      ];
+
+      // return MerchandiseItem
+      const merchandiseItem = {
+        id: item.id,
+        label: item.name,
+        gender: item.sex,
+        genderOptions,
+        size: item.size,
+        sizeOptions,
+        description: item.description,
+        images,
+      };
+      logger?.debug(
+        `Created item <${JSON.stringify(merchandiseItem, null, 2)}>`,
+      );
+      return merchandiseItem;
+    });
+
+    logger?.info(`Computed <${items.length}> merchandise items`);
+    return items;
+  };
+
+  /**
+   * Transform merchandise data into MerchandiseCard format
+   * grouped by gender and then by name of the product.
+   * This is used to display merchandise cards with composite data
+   * showing information about all variants of the product of that name.
+   */
+  const computeMerchandiseCards = (): Record<Gender, MerchandiseCard[]> => {
+    logger?.info('Computing merchandise cards by gender');
+    logger?.debug(`Transform <${merchandise.value.length}> items to cards`);
+
+    // group merchandise by gender
+    const merchandiseGroupedByGender = merchandise.value.reduce(
+      (acc, item) => {
```

Comment:
better var name `aac` (I don't know what does it mean 'aac')

Reply:
Yes, `acc` is an abbreviation for "accumulated value", but the name should be clearer.
Renamed to `groups` + refactored the reduce function to standalone `groupMerchandiseByKey` function (based on later comment).